external-groups/tool/src/main/java/uk/ac/ox/oucs/vle/MappedGroupsResource.java

```diff
@@ -108,7 +108,8 @@ public Response addMappedGroupToSite(@FormParam("group") String group, @FormPara
 					log.debug("Set site : "+ site.getId()+ " provided id to: "+ providedId);
 				}
 				siteService.saveSiteMembership(site);
-				
+				site.getAuthzGroupService().refreshAuthzGroupInternal(site.getAzg());
```

Comment:
Doesn't this result in a double save of site memberships, as you fire off one straight away and it also gets put onto the background refresh thread?